runtime/io.c

```diff
@@ -848,6 +849,21 @@ CAMLprim value caml_ml_output(value vchannel, value buff, value start,
   return caml_ml_output_bytes (vchannel, buff, start, length);
 }
 
+CAMLprim value caml_ml_output_bigarray(value vchannel, value vbuf,
+                                       value vpos, value vlen)
+{
+  CAMLparam1(vbuf);
+  struct channel * channel = Channel(vchannel);
```

Comment:
You probably also need to register `vchannel` as a root here.

Reply:
Done. Thanks.